### PR TITLE
arch: Remove the duplicated up_tls_info implementation

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Define up_tls_info in arch/arch.h directly if the general one isn't suitable

## Impact
Code refactor, no real change

## Testing
Pass CI and ostest

### CODE REVIEW

File: include/nuttx/arch.h [hunk 1]
```diff
@@ -1791,10 +1791,18 @@ int up_timer_start(FAR const struct timespec *ts);
...
- * actual implementation may be a MACRO or an inline function.
+ * The actual definition is provided in arch/arch.h as a macro. The default
+ * implementation provided here assume the arch has a "push down" stack.
  */
 
+#ifndef up_tls_info
+#  ifdef CONFIG_TLS_ALIGNED
+#    define up_tls_info() TLS_INFO((uintptr_t)up_getsp())
+#  else
+#   define up_tls_info() tls_get_info()
```
Comment (gustavonihei) [inline include/nuttx/arch.h:1802]:
```suggestion
#    define up_tls_info() tls_get_info()
```

Comment (xiaoxiang781216) [reply include/nuttx/arch.h]: Done.

**Comment by xiaoxiang781216** [comment]
Need PR https://github.com/apache/incubator-nuttx-apps/pull/912 to pass the CI, @gustavonihei .

**Review by gustavonihei — APPROVED** [review]
LGTM

**Review by Ouss4 — APPROVED** [review]

**Pull request merged**